collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,346 @@
+
+This release also includes a security fix, previously disclosed in the 4.0.4 RC1 blog post. If your project uses `ENetMultiplayerPeer` or the low-level `ENetConnection`, there is a potential denial of service attack that utilizes a flaw in the ENet library. We strongly recommend updating to versions 4.0.4, 4.1.2, and 4.2 (once it is released as stable). Thanks to [Facundo Fernández](https://github.com/Facundo15) for reporting this vulnerability. A patch has been submitted to the upstream ENet repository as well.
+
+These multiplayer changes were contributed by Godot's networking and Web platform maintainer [Fabio Alessandrelli](https://github.com/Faless).
+
+
+### Porting and platforms
+
+Godot prides itself on first-class support when it comes to porting, including all mainstream desktop platforms, mobile platforms (with some exciting news [coming to C# users](#c--net)), and the Web platform.
+
+Part of this support includes better integration with the host environment. In this release Godot's porting guru [Pāvels Nadtočajevs](https://github.com/bruvzg) is adding native file selection dialogs for Linux, macOS, and Windows, allowing you to leverage familiar user interface in your games and apps ([GH-47499](https://github.com/godotengine/godot/pull/47499), [GH-79574](https://github.com/godotengine/godot/pull/79574), [GH-80104](https://github.com/godotengine/godot/pull/80104)). This feature is especially important for sandboxed apps on macOS.
+
+Another extremely useful interaction with the operating system is copying and pasting images. Thanks to efforts from [Deakcor](https://github.com/deakcor) this is now implemented for macOS and Windows ([GH-63826](https://github.com/godotengine/godot/pull/63826)), but Linux/X11 should hopefully follow suit soon ([GH-81439](https://github.com/godotengine/godot/pull/81439)).
+
+#### Android
+
+Platform integration is often a compromise allowing you to marry two completely foreign technologies for a seamless experience. Godot's Android platform architecture currently contains one such compromise, being based on the Android Fragment component that significantly limits our control over the Godot application running on that platform. Over the course of several months project's lead Android maintainer [Fredia Huya-Kouadio](https://github.com/m4gr3d) worked tirelessly on refactoring the platform integration to solve this limitation at its core.
+
+New Android platform architecture (introduced by [GH-76821](https://github.com/godotengine/godot/pull/76821)) decouples Godot's application from the Android Fragment component, unlocking new features in the process such as multiple windows and quicker startup times for Godot apps and games. As a continuation of this work, the Godot Android plugin framework has been updated as well ([GH-78958](https://github.com/godotengine/godot/pull/78958), [GH-80740](https://github.com/godotengine/godot/pull/80740), [GH-81368](https://github.com/godotengine/godot/pull/81368)).
```

Comment:
It's tricky, because in this case it's not talking about applications as a user-facing concept, but about the "Godot application" part of the distribution, as opposed to the boilerplate/platform hooks.

```suggestion
New Android platform architecture (introduced by [GH-76821](https://github.com/godotengine/godot/pull/76821)) decouples Godot's application part from the Android Fragment component, unlocking new features in the process such as multiple windows and quicker startup times for Godot apps and games. As a continuation of this work, the Godot Android plugin framework has been updated as well ([GH-78958](https://github.com/godotengine/godot/pull/78958), [GH-80740](https://github.com/godotengine/godot/pull/80740), [GH-81368](https://github.com/godotengine/godot/pull/81368)).
```

Reply:
Ah thanks, I wasn't sure about that.